Dockerfile

```diff
@@ -1,15 +1,17 @@
 FROM fedora:latest
 MAINTAINER "Daryl Metzler"
 
-ADD . /opt/charge
+RUN dnf -y install ruby rubygem-bundler ImageMagick pngquant file
 
 WORKDIR /opt/charge
 
-RUN dnf -y install ruby rubygem-bundler ImageMagick pngquant file \
-   && bundle install --without test development
+COPY Gemfile .
+COPY Gemfile.lock .
+RUN bundle install --without test development
 
-ENV CHARGE_PORT=8881
+COPY . .
```

Comment:
What does `COPY . .` do?

Reply:
It moves the current directory into the WORKINGDIR.

Reply:
Read: it copys all of the code in the repo into the container image we're building.